### PR TITLE
feat(balance): Make fighters go into a new "wrecked" state

### PR DESCRIPTION
**Pull request by danaris**
Introduce a new "wrecked" state for the player's fighters and drones, that they will enter upon taking enough damage to be fully destroyed.

This will prevent them from taking any further damage, and when their carrier lands on a planet with a shipyard, they will be repaired automatically.

----------------------
**Feature:** This PR implements the feature request detailed and discussed in issue #3920
Also related: #4034, #906

## Feature Details
Upon being reduced to 0 hull, fighters and drones (henceforth just "fighters", for simplicity) belonging to the player will, mechanically, have their hull set to 0.1 and go into a "wrecked" state. 
Wrecked fighters will no longer take damage of any kind, and act as "disabled" for most purposes. 
The AI will ignore these ships for targeting purposes, _except_ that they will ask their parent ship for help (if they have no parent ship, they will instead ask any escort with bays for help; if there are none, they're screwed anyway).
In particular, other escorts will no longer go to wrecked fighters to attempt to repair them.
For the flagship, when Boarding a wrecked fighter, it will simply pick it up into the appropriate bay.
Wrecked fighters in bays will not be repaired.
Upon landing on a planet with a shipyard, all wrecked fighters will be immediately repaired, and a dialog indicating this displayed.

This is intended to be a (partial) solution to the issues with fighter balance and frustration. I have read through several past discussions on the topic, and at least thus far have not seen whether there is a real consensus on which of the various suggestions should be implemented, so I hope that this one is seen as reasonable.

## UI Screenshots
Repair dialog:
wrecked-fighter-repair

Otherwise, it just looks like the fighters are disabled.

## Usage Examples
N/A

## Testing Done
Bought an Ibis and some Petrels, manually shot down a Petrel several times and observed the behavior of escorts. 
With the Ibis as my flagship, boarded the wrecked Petrel, successfully loading it aboard, then landed, successfully repairing it.
With the Ibis as an escort, shot down the Petrel and observed the Ibis successfully load it automatically.

### Automated Tests Added
I am, I fear, unfamiliar with automated tests. If this requires some, I will happily take direction, or studiously observe someone else's additions there for future reference.

## Performance Impact
Adds a few (fairly simple) tests to `AI::AskForHelp()`, `AI::FindTarget()`, and `Engine::DoCollisions()`. Adds some checks in `Ship::Move()` that will, for wrecked fighters, disable some methods (such as `DoGeneration()`), thus likely slightly reducing load.

### CODE REVIEW

**Comment by Saugia** [comment]
I don't think making fighters incapable of exploding, therefore making them different than all other ships by the fact that they cannot die, is a plausible way to balance them. Fighters are not just secondary thoughts to your ships or carriers, they too are spaceships in space fighting, and shouldn't get special treatment just because they are small. They should be treated like any other spaceship and are susceptible to fire and dying. This isn't like some RTS games where fighters just make up a ship's compliment as if they aren't ships themselves, and are more of a group/secondary aspect. If we are going to balance fighters, it would be in their purpose, role, stats, etc.., not making them theoretically invincible.

I'll leave this open for discussion, but this has been proposed in the past and wasn't particularly liked.

**Comment by danaris** [comment]
Indeed; in the previous discussions, this suggestion did not garner a majority of support.

However, neither did any of the other related suggestions.

The problems with fighter-based play remain, and I, personally, am skeptical that an approach that insists fighters remain at absolute parity with all other ships will be able to address them. 

There **is** a fundamental difference in the gameplay of a Sparrow in the early game, whether as the player's flagship or a very early escort, and the gameplay of a Barb, or even a Petrel, in the late game, as part of a complement of fighters and drones. 

- In the former case, the player is likely to be choosing battles fairly carefully, trying to pick isolated targets, and the main plot will not be pushing the player to take on significant combat challenges. 
- In the latter, the player is much more likely to be in a position where they **want** to be able to tackle stronger enemies, even larger fleets of enemies, and the plot will be throwing them into situations where they are required to do so.

The overall **balance** of fighters vs other ships is something I do not feel qualified to address: matters of speed/acceleration, allowing unique outfits or restricting to only certain outfits, adjusting shields, etc all require a level of mastery over the game's systems that I definitely do not have (not to mention lots of math). However, even were all that to be ironed out, the fact is that, compared to other types of ships in the late game, fighters are very fragile, and losing fighters, especially those the player has spent time and credits customizing, every time you go into a nontrivial battle is Not Fun.

The two basic options for dealing with that (at a high level—not attempting to be exhaustive) are to make fighters less customizable, or to make them less disposable. Personally, I like the ability to customize fighters—to treat them, in fact, much like any other ship in all ways but one. 

Thus, while I do not particularly intend to put a cat among pigeons, I **would** like to see some kind of resolution on the question of fighter disposability, because I rather enjoy fighter-based combat, and want to be able to do so without the endless frustration of losing half my fighters every other engagement.

I hope that this helps to explain why I chose to make this my first PR.

**Comment by Saturnalia9** [comment]
> I don't think making fighters incapable of exploding, therefore making them different than all other ships by the fact that they cannot die, is a plausible way to balance them. Fighters are not just secondary thoughts to your ships or carriers, they too are spaceships in space fighting, and shouldn't get special treatment just because they are small. They should be treated like any other spaceship and are susceptible to fire and dying. This isn't like some RTS games where fighters just make up a ship's compliment as if they aren't ships themselves/ are more of a group/secondary aspect, they are full-fledged spacecraft as is any other. If we are going to balance fighters, it would be in their purpose, role, stats, etc.., not making them theoretically invincible.
> 
> I'll leave this open for discussion, but this has been proposed in the past and wasn't particularly liked.

I'm more partial to this solution than to the other more popular solution of making fighters just missile outfits. We've talked about this subject for years now, and no solution has really gone past the discussion phase. So even if the PR is unlikely, it's at least already present.

**Comment by Saugia** [comment]
Yes, I agree that the missile outfit is probably the worst way to go about this, and would definitely take this over it. However, I think this is worth a more in-depth discussion before we consider this as the only "good" option for balancing "repairing" fighters. I think the last thing we should be doing, next to turning them into an outfit, is labeling them with an "invincibility tag". As I said, this is not an RTS, and we shouldn't treat fighters like they are an outfit or secondary object to any other ship. 

I know people will disagree, but fighters *aren't* as bad as everyone makes them out to be. The are quite strong, especially in the early game, and especially considering the game is based around solo-ship play. Now, that isn't necessarily what this PR is attempting to propose, but it's important to take that into account. If fighters are suddenly invincible and no longer require replacement, that effectively *does* turn them into a sort of outfit, actually. It all depends on how we want to envision fighters in Endless Sky, hence why this warrants a more specific discussion, and no, that isn't various smaller discussions over the years, or a channel in discord, etc.. 

It's something we should sit down and make a concrete decision on as to what we think is most appropriate for the game. Perhaps that is doing this PR. If it's what's best for the game, then I'd be fine with it. However, as of right now, I am not convinced. I very much dislike the solution to fighter repair being an invincibility tag. I don't think that the hassle of repairing them quite equates to making them straight up invincible and removing the requirement to re-purchase them entirely.

**Comment by danaris** [comment]
Would it be more palatable if repairing the fighters at a shipyard had a cost? I haven't poked at actually doing this, but I would imagine it would be fairly easy to determine the total cost of the fighter and require a payment that's some percentage of that (possibly even an adjustable percentage, depending on factors TBD) before repair can take place.

**Comment by Saugia** [comment]
I am under the impression that is a given if we do take this route. Though again, I would like to see some more in-depth discussion about this with the dev team and other content creators. I'd actually would prefer to wait until we did an overall balance on fighters, speed, etc., and then see if it's still worth considering the repair requirements for fighters.

**Comment by AvianGeneticist** [comment]
I really love this; fighter balance has been talked about for aeons so to finally see some concrete movement is great to see.

A couple notes:
Repairing fighters should still cost some money. Maybe a third (or more, I'd be willing to pay full price (or even more) for reasons below) of their cost or so. This would still make it so fighters are a bit more of an involved decision (much like using ammunition) on the player's part, while not making it so a player has to trek all across the galaxy because, whoops, one of their uber-customized Terns got blown up. This way, it comes as a compromise, where liberally dispersing fighters will cost you some time (boarding to recover) and money, but it's not nearly as huge of a timesink as having to go harvest [loot-only outfit] for 2-3 hours per fighter.
The dialog popping up every time will probably get annoying. You can probably get away with one (the first time it happens, like the tutorials for fleet mining), since the player will see that their fighters have been repaired when they get up and leave.

**Comment by TheOnlyKnight** [comment]
I think this is our best option in the first place. Balance or not, the reason I don't use fighters is the replacement time, not the cost or usefulness. They're already fairly powerful, but terribly fragile. If I lose a fighter, I either have to carry a Deep River around with all my outfits for it, or spend 15 minutes traveling the galaxy to replace the fighter itself and all the outfits it used. If some portion of that cost, maybe even the full 100% non-depreciated value of the fighter, was frontloaded to the first planet I land on (maybe require one with an outfitter?) after "losing" one, I would use them much more.

I think it's useful to explicitly say what I think the problem with fighters is from a gameplay perspective - that no one uses them. I think that's the same perspective this PR is working from, since it focuses on the player's fighters.

If the problem we're trying to solve is _the non-use of **customizable fighters** due to horribly inconvenient replacement time_, then allowing them to be completely "repaired" for a cost is our best option. If the problem is something more general with the _role_ of fighters, then maybe not, but I don't see that being the case.

**Comment by Saugia** [comment]
Don't get me wrong, I do believe some sort of balance is still warranted for making fighters viable due to repairing and replacement, but I reiterate that having an invincible tag is, in my opinion, not the right call. If I were to say a preference, I would much prefer the alternative of having fighters "rebuild". Or, frankly, I would choose an option where you lose the fighters, but they are replaced. My biggest issue is having these separate-status fighter craft flying around invincible and/or not targetable by weapons as it's far different from anything else in the game and can possibly cause confusion. We should be avoiding having any mechanics regarding a ship's hull that are exclusive or otherwise not found anywhere else in the game, it is too inconsistent and starts to push the game in a direction I believe it doesn't represent.

**Comment by TheOnlyKnight** [comment]
Use this mechanic, but instead they turn invisible and RTB. Pay 100% cost on landing to "rebuild" them, as you say.

**Comment by Saugia** [comment]
Invisibility is even worse. The issue as I said is having a fighter go into some exclusive state after "dying".

But, I've said my piece for not being a fan of this PR. I'll leave it to discussion, and it will probably have to wait for consideration until we actually get around to looking to it, as with all other PR's. Fighter balance isn't on the table quite at the moment, but something I would like is to have a balance-focused patch where we can have all of this mechanical discussion of the game's balance. It probably won't be the next 2 patches or so though. I don't think a single PR is "a concrete way forward", since it's been a multitude of times. I respect the interest in it and am happy people like the concept, I just want to clarify that this isn't some magical "fix fighters" PR that if we do like it, fixes them, and if we don't like it, is the end of the world for fighters.

**Comment by Hurleveur** [comment]
The way I see it, conceptually, this allows you to get the wreckage of fighters from the battlefield to allow repairing them even if there isnt a shipyard in the system, whilst still having to collect them.
Meaning that there is more player engagement required than just a "rebuild" dialog in the outfitter.

I'd put the price at 50%, and make a graphical change to show they become wreckage. Then it makes sense they dont get shot at (although mb we should consider it and make the repair cost higher? Or just eventually destroy them ig)

As to that behavior being different from the rest of the game I wouldnt mind it with other ships: they could give wreckage upon dying, except you'd just sell this one? Now it could be a perf issue if they were to all blow into too many pieces, too, but should be fine.

This also probably sould be a togglable option

**Comment by Arachi-Lover** [comment]
Perhaps make it so it's only a % chance they enter this state? Like 70/30 for wreckage/destroyed.

**Comment by Saturnalia9** [comment]
> Yes, I agree that the missile outfit is probably the worst way to go about this, and would definitely take this over it. 
>
There was another suggestion about adding evasion chance to fighters, but that was argued down the most. I also don't want such blatant RNG in this game.

Another suggestion is lateral engines, but I really don't think that would help nearly as much as any of the other suggestions. 

> I very much dislike the solution to fighter repair being an invincibility tag. 
>
It's not so much that they're invincible. They can be picked up and repaired at a spaceport, but until then they're still essentially dead/useless ship hulls.

**Comment by TomGoodIdea** [comment]
If we had a good AI allowing fighters to run away when damaged (making use of their high speed), they would be destroyed less often.

**Comment by Hurleveur** [comment]
They already run away when damaged, but thats too late, they probably should run when targeted at all

**Comment by TomGoodIdea** [comment]
> They already run away when damaged, but thats too late, they probably should run when targeted at all

That's true, I forgot about it.
Actually I think it would be better to raise the threshold than check if enemies are targeting you, since how would you know that lore-wise?

**Comment by TheOnlyKnight** [comment]
TL;DR, fighters are placed such that special treatment to "improve" them is entirely warranted. A PR that changes nothing about fighters' _effect_ but improves their _use_ for the player is the best, and as far I have seen, the only solution for the lack of their use in the average player's toolkit.

Fighters DO disengage after a certain amount of damage, and return to their carrier. They'll even keep moving to their carrier if their own regeneration takes them back up above the disengagement threshold. The thing is, they're still so fragile that by the time they've turned around, they've lost the rest of their health past the threshold and die.

As I've said on the Discord, I don't think any solution - be it a _strategy_ in the status quo or a PR to _change_ the status quo - that gets less than a 100% "recoverable" rate for fighters is palatable. If I lose a single fighter, and there's no better way to replace it than we have now, that means I have to push a crusade/pilgrimage hybrid across the galaxy buying plundering all the outfits I want for my loadout.

Just like ammunition, players are presented with three options:
1. Don't use the mechanic, because replacing costs are too time-consuming.
2. Use the mechanic and regularly restock to keep the fleet at tip-top performance.
3. Use the mechanic and restock as late as possible to save time on trivial shopping trips.

For a lot of people, No. 1 is most attractive - they don't like adding more running costs to an already crew-heavy fleet, so they stay away from it. This is true even for basic ammo weapons, but it's even more relevant for fighters, because on top of the sizable cost of the fighter itself, there's the extraordinary time cost of buying and plundering the outfits used to make them.

For those who do use fighters, they're incentivized to play with No. 3, because while the monetary cost will always go up, the time cost remains the same for purchasable, so they can very easily "take the L" and lose fighters across multiple engagements, thus replacing more than the minimum per outfitting pilgrimage. This lends itself to the risk of being marginally under equipped for an engagement, taking more losses than perhaps necessary. And, they player will always eventually have to bite the bullet and re-stock and re-outfit their fighters, unless they take the easier option and just... don't, sell their carriers, and continue with the brainless strategy we're trying to provide alternatives to - the HW spam of the highest calibre.

I see a lot of resistance to this sort of change come from the idea that, as I understand it, fighters don't warrant special treatment. The idea that the game should change as a whole to naturally equalize the use of different classes instead of shoe-horning in some certain traits and others none. In this way, it would avoid a terrible future where we've brashly buffed everything besides HWs and also nerfed HWs into oblivion in order to promote more varied playstyles, in the process causing new players to wonder why everything seems so awkward and arbitrary. I do not disagree with this sort of naturalistic game design.

However, I do not think it will fix fighters. This nice, elegant idea of game design is amazing if you want to make, say, HWs stop doing the jobs of LWs but with more shields and outfit space, but that isn't our problem here. We're looking at Fighters vs no fighters, not fighters vs HWs or LWs. Our problem isn't that fighters are weak, or that they're to expensive in credits to use. Our problem is that fighters are a _powerful_ but _completely useless_ tool, because the vast majority of players will look at the time cost of losing and replacing a single fighter and decide not to pay that cost, at least in a legitimate way.

In any future wherein our problems with Light to Heavy Warships are solved, this problem with fighters will remain, because fighters have their own valid, effective use that is currently inconvenient. The PR here solves this problem perfectly - fighters are not buffed, per se, because an engagement with or without this PR will go the same - fighters will still be disabled and rendered non-threats, swarms will still evaporate isolated targets - but, when the _player_, the party not using fighters that we want to be using fighters more, when the _player_ looks at their options, and they see the credit-heavy course of carrier fleets, they aren't deterred from the much, much worse downside of a replacement crusade that is boring and repetitive, especially for outfits that can't be bought from planets.

**Comment by nylki** [comment]
I am hesitant to sidetrack the PR with yet another idea, but seeing the controversy around fighter mechanics, maybe this could be a conpromise:

What about having special flotsam that are created when a fighter explodes, which the flagship can collect after a fight and gets a discount when re-buying fighters at port when using that flotsome as currency. (Eg. Damaged engine part flotsam)

**Comment by Hurleveur** [comment]
> > They already run away when damaged, but thats too late, they probably should run when targeted at all
> 
> That's true, I forgot about it. Actually I think it would be better to raise the threshold than check if enemies are targeting you, since how would you know that lore-wise?

The orange of enemies turns to red when they target you (or your whole fleet? Not sure for that)

**Comment by TomGoodIdea** [comment]
> The orange of enemies turns to red when they target you (or your whole fleet? Not sure for that)

it's for fleet:
https://github.com/endless-sky/endless-sky/blob/f83b8596f7fba44c194998cd9afe220fea100369/source/Engine.cpp#L98-L99

**Comment by Saturnalia9** [comment]
> If we had a good AI allowing fighters to run away when damaged (making use of their high speed), they would be destroyed less often.

It already exists, and has been implemented for a while now. One problem is that there's very little diversity in speeds ranging from fighters to heavy warships, which makes fighters extremely vulnerable. QuantumShark (DoomKorath in Discord) has a PR #8842 that addresses mobility differences.

The other problem is that jousting is the only combat method currently (and likely will stay that way for a long time). Fighters will joust with heavies, and by the time they get to that threshold of "oh no, I need to run", they're already essentially in the middle of the joust. A millisecond after the AI realises it needs to run, it's already facetanking particles/protons in melee range.

**Comment by Saturnalia9** [comment]
> I am hesitant to sidetrack the PR with yet another idea, but seeing the controversy around fighter mechanics, maybe this could be a compromise:
> 
> What about having new kind of flotsam that is created when a fighter explodes, which the flagship can collect after a fight. 
> When re-buying fighters at port you can use that fighter-specific flotsam to get a discount. (Eg. damaged engine part flotsam, damaged shield generator flotsam, etc., or just generic fightername-xy-damaged-parts-flotsome)

That's essentially this PR. He already said he'd not mind adding a 50% cost to repair the fighter (which I prefer over the free repair).

The difference is that you're mentioning buying a new fighter (or so it reads), while the PR is specifically repairing the old fighter.

**Comment by nylki** [comment]
> > I am hesitant to sidetrack the PR with yet another idea, but seeing the controversy around fighter mechanics, maybe this could be a compromise:
> > What about having new kind of flotsam that is created when a fighter explodes, which the flagship can collect after a fight.
> > When re-buying fighters at port you can use that fighter-specific flotsam to get a discount. (Eg. damaged engine part flotsam, damaged shield generator flotsam, etc., or just generic fightername-xy-damaged-parts-flotsome)
> 
> That's essentially this PR. He already said he'd not mind adding a 50% cost to repair the fighter (which I prefer over the free repair).
> 
> The difference is that you're mentioning buying a new fighter (or so it reads), while the PR is specifically repairing the old fighter.

Well, you are right, its similar.
The relevant difference I propose is to create _flotsam_ instead of having a new disabled ship status. Floatsam is already an established concept in the game and has certain visual clues related to destroyed objects, which would fit here in my opinion. There is already a mechanic in place for collecting flotsam, selling it etc. (other ships could collect it as well). It just might be easier to integrate with existing mechanics and to expand on it.

Just my two cents. Do with the idea what you will 🙂

**Comment by Hurleveur** [comment]
The flotsam will likely be too hard to catch, although it's good to expand an existing idea for sure

**Comment by Saturnalia9** [comment]
Probably harder to create special floating flotsam associated with a ship than to create a new status for ships. Floating flotsam currently is only associated with outfits (both regular outfit and trash "outfits").

**Comment by Quantumshark** [comment]
I do think this is a sensible idea.
The cost and inconvenience of replacing and reoutfitting fighters are downsides that only exist for the player, not for NPCs. That creates a problem for balancing fighters - if we make fighters strong enough in the player's hands to merit those downsides, they'll be completely overpowered when used by NPCs. If we balance them around their effectiveness when used by NPCs, they'll never be worth using for the player.
With that in mind, it becomes clear that we need a solution that either mitigates or removes those player-specific downsides, or adds some additional advantage for fighters that only exists for the player, not for NPCs. This PR does seem like a reasonable approach to that.

File: source/AI.cpp [hunk 1]
```diff
@@ -3583,11 +3601,11 @@ double AI::RendezvousTime(const Point &p, const Point &v, double vp)
...
 	// p.x^2 + 2*p.x*v.x*t + v.x^2*t^2
-	//    + p.y^2 + 2*p.y*v.y*t + v.y^2t^2
-	//    - vp^2*t^2 = 0
+	//	+ p.y^2 + 2*p.y*v.y*t + v.y^2t^2
+	//	- vp^2*t^2 = 0
 	// (v.x^2 + v.y^2 - vp^2) * t^2
-	//    + (2 * (p.x * v.x + p.y * v.y)) * t
-	//    + (p.x^2 + p.y^2) = 0
+	//	+ (2 * (p.x * v.x + p.y * v.y)) * t
+	//	+ (p.x^2 + p.y^2) = 0
```
Comment (Quantumshark) [inline source/AI.cpp:3608]: This looks like a mistake - why are you adding whitespace in a comment about a section that you aren't changing?

File: source/PlanetPanel.cpp [hunk 2]
```diff
@@ -90,6 +90,33 @@ void PlanetPanel::Step()
...
+		if(repairedFighters > 0)
+		{
+			ostringstream out;
+
+			string fighterWord = "fighters";
+			if(repairedFighters == 1)
+			{
+				fighterWord = "fighter";
+			}
+			out << "You quickly see your wrecked " << fighterWord << " transported to the shipyard for repair.";
```
Comment (Quantumshark) [inline source/PlanetPanel.cpp:115]: It's a minor thing, but it'd be nice if this correctly identified whether it was fighters, drones or both that had been repaired, and updated the text accordingly.

File: source/AI.cpp (hunk 1 above)
Comment (danaris) [reply source/AI.cpp:3608]: ....oh, whoops. I think that was an accidental consequence of doing a global search & replace for <4 spaces> to <tab>, because I'd made enough edits to AI.cpp before I realized my IDE was still defaulting to spaces that it seemed quickest.

If it's desired, I'm happy to revert that change.

File: source/PlanetPanel.cpp (hunk 2 above)
Comment (danaris) [reply source/PlanetPanel.cpp:115]: An excellent point. 

What's the most canonical way to tell? Should it test for whether each ship that got repaired has any max crew? Should it use the `category`? Or is there a better way?

Comment (Hurleveur) [reply source/PlanetPanel.cpp:115]: We dont have much code linked to the names of the category yet,do we?
I'd check for crew and use "unmanned drone" and fighter, but mb its hacky

Comment (Quantumshark) [reply source/PlanetPanel.cpp:115]: In this case, I think just matching `category` is probably the best approach. If you report the name of the category directly, you would also have some degree of compatibility with categories added by plugins, though I don't think there'd be a way to support nonstandard pluralisations.

**Comment by Pshy0** [comment]
Shots could pass above disabled fighters without hitting them. They are small after all, you should be able to aim at targets behind them.

This would reduce their chances of being destroyed. They would still be hit if specifically targeted, from explosion of other ships, or by projectiles that were already shot at them.

**Comment by Hurleveur** [comment]
That would mean they wouldnt die at all if the enemy disables since the target is changed when the fighter is disabled?

**Comment by danaris** [comment]
Given the frequency of blast damage, both from secondary weapons and from dying ships, I suspect they'd still get destroyed at a pretty high rate with that change, especially in the kind of grand mêlée where you'd most want to be able to use them.

**Comment by Saturnalia9** [comment]
As it stands currently (and in the near future), I'd never want to deploy my fighters, simply because anything can 1v1 a fighter and win, aside from other fighters and drones.

And, I don't speak for everyone, but I'm too scarred from their current weakness to immediately use fighters until I get more confidemt with them. 😅

**Comment by TheOnlyKnight** [comment]
Okay, one problem with the PR itself is that wrecked fighters have no drag. It's hard to dock and bay them when they're flying off to Timbuktu faster than the carrier's top speed.

**Comment by danaris** [comment]
> Okay, one problem with the PR itself is that wrecked fighters have no drag. It's hard to dock and bay them when they're flying off to Timbuktu faster than the carrier's top speed.

Ah, I _thought_ it seemed like they were much faster to drift than regular disabled ships. Knowing that I just need to re-enable drag for them should make it pretty easy to fix.

**Comment by Zitchas** [comment]
Just adding my two cents: I like the idea of fighters getting "wrecked" instead of destroyed; and subsequently being retrievable and repairable in a shipyard. I'm not super fond of this being restricted to the player's fighters, though, nor am I particularly fond of them actually being invincible. If I (or NPCs) really want to go out and really make sure those fighters are good and destroyed, I think they should be able to.

Some ideas:
1. Rather than having wrecked fighters be invincible, having them get a degree of damage reduction (say, 50% hull damage reduction) while they are in their wrecked state would mean that they'll still get toasted by inopportune area effects and/or direct fire now and then; but it will still let them live a *lot* longer than normal. Given how weak fighter's hulls are, maybe 75% damage reduction. From a lore perspective, two things become true once a fighter is "wrecked": a) the fighter is generally no longer targeted by weaponry, so even if they do hit, they're not typically being expressely targeted for maximum impact anymore; and b) they no longer have active systems, so damage is much less likely to be lighting up fuel lines or short circuiting live circuits, etc. and thus causing less cascading damage.
2. Have wrecked fighters be ignored by everyone except the "Vindictive" personality. So Vindictive NPCs will still try to annihilate fighters, but the rest should ignore them.
3. I rather like the idea of being able to collect my enemy's wrecked fighters if I managed to eliminate their carrier. Gives an alternate mechanism for building up a carrier fleet.

**Comment by Zireael07** [comment]
2&3 are good points.

**Comment by lumbar527** [comment]
> 3. I rather like the idea of being able to collect my enemy's wrecked fighters if I managed to eliminate their carrier. Gives an alternate mechanism for building up a carrier fleet.

Like in EVN? (Haven't encountered it; only read about it so it could work differently than I think)

**Comment by Saugia** [comment]
In EVN you still have to disable them. You still technically capture them and can call them into your bay, but what I think Z is saying is if you do capture it, if you have a bay slot open, it immediately goes into the bay rather than waiting for your command.

**Comment by Saturnalia9** [comment]
> 1. Rather than having wrecked fighters be invincible, having them get a degree of damage reduction (say, 50% hull damage reduction) while they are in their wrecked state would mean that they'll still get toasted by inopportune area effects and/or direct fire now and then; but it will still let them live a *lot* longer than normal. Given how weak fighter's hulls are, maybe 75% damage reduction. 

So untargetable and only able to kill them with AOE, basically? 👍

> 2. Have wrecked fighters be ignored by everyone except the "Vindictive" personality. So Vindictive NPCs will still try to annihilate fighters, but the rest should ignore them.

How often does the player find vindictive? I know the major human governments have it, including Remnant I think, after they finish looting (still human btw).

**Comment by TheOnlyKnight** [comment]
As it is now, the PR leaves wrecked fighters at very low health, so I don't think damage reduction would help them remain after being reduced for 0 HP the first time. I really like the concept of having vindicative AI hunt down wrecked fighters - it should reduce clutter in places like the Far North where Aeries show up a bunch, while still allowing the player to capture those fighters.

I'm not sure on having just stray fire kill wrecked fighters, because there wouldn't really be any difference with or without the PR - fighters go in between friendly and hostile HWs, get disabled, and die in the crossfire before the kiting death spiral migrates somewhere else. I'm uncomfortable with AoE damaging them for the same reason, but I don't think it's as problematic as stray fire, and incetivising the player to avoid fielding fighters against ostensibly fighter-oriented loadouts is _that_ much of a problem.

I keep brainstorming alternative ideas, but every one of them runs into _some_ amount of fighter attrition, which presents a replacement problem that only the player experiences.

**Comment by Saugia** [comment]
Ultimately I don't feel that a direct approach is the best way to solve the issue with fighter replacement, especially through excusive or unique mechanics. In Endless Sky, we do not give the player any statistical benefits that we wouldn't also give to npcs. Npcs and the player are on even ground when it comes to ships and outfits, and the difference should be made up with the player's skill and usage of them, rather than giving the player and advantage that Npcs do not have.

I think more indirect adjustments, such as the speed/turn proposals for larger ships, or perhaps nerfing turret traverse (which is something we've done in the past) are better ways to go about this to keep the scope to the entire game, and not specifically fighters or the player. Any buff we give to only fighters doesn't help other ships that are also in need of adjustments (such as interceptors), which we should look at more overall, rather than specific case.

**Comment by Amazinite** [comment]
Adding to Saug's comment, a central balancing point of the game is that given some amount of credits, a larger number of smaller ships should (and often does) beat out a smaller number of larger ships. So say you have one heavy warship and however fighters equals the worth of that warship, the fighters almost always win, usually unless the heavy warship has a very particular loadout for combating fighters.

Now the trouble that this makes for the player is that the player, unlike the NPCs, has to worry about rebuying lost fighters. So the main thing we should be trying to do, which this PR does do, is make fighters more likely to survive combat. Notably, this is not the same thing as making fighters last longer in combat. To this end, and with keeping in mind the point that Saug has mention of not giving the player special treatment, I'd prefer to the the development of a mechanic where disabled ships are able to be given a short but powerful kick of thrust. Imagine an "emergency thruster" that could be given to all fighters. Once a fighter becomes disabled, this emergency thruster would kick the ship backward from its current facing angle, hopefully out of combat. This way the player is able to retrieve their fighters after combat is finished without having increased the combat capability of the fighters.

**Comment by Saturnalia9** [comment]
From what I read, this PR forces you to go back to port to repair the fighter at a price. 🤔

**Comment by danaris** [comment]
Indeed; this PR explicitly—and intentionally—did _not_ allow for repairing and redeploying within the same combat. That changes within-combat balance, rather than just long-term balance, which is what I was trying to avoid.
